### ISSUE TITLE
an Auto Incrementing Save

### ISSUE THREAD

**Issue by allaze-eroler** [aseprite/aseprite#2765]
i would like to have an an Auto Incrementing Save as feature so i wouldn't have to accidentally save my work in flattened layer. it's the forth incident that happened to me in the past years...

many art software use that method, it's particularly true when there is a complex work to deal with.

the idea is that the incremental save files would be saved in a separate folder from the pixel art project will be a list of the same project but in different state of progress:

foo_v-001.aseprite
foo_v-002.aseprite
foo_v-003.aseprite
foo_v-004.aseprite
foo_v-005.aseprite

in the end of the pixel art project, the artist could choose to continue to keep them or delete the folder for good if they never plan to continue working on it.

what do you think of this idea?

**Comment by dacap** [comment]
Hi @allaze-eroler, if you have your `sessions` folder, we might be able to recover your sprite before the flatten command (we save 3 versions in the data recovery session). It's a difficult process but it's something to do as a last resource. You can send it to support@aseprite.org

About the auto-save feature, I have two thought:
1. We could add an option to auto-save each file, the only problem is that the operation gets slower when the file gets bigger (so might not be a generic solution).
2. Other possibility would be to improve the data recovery information to save more versions and add the possibility inside the program to navigate through different versions of the sprite (e.g. latest 20 versions).

I think the 2nd point is doable.

**Comment by allaze-eroler** [comment]
Yes, that is what i did but turn out it was too late to get back these layers... based on what you said, increasing the number of version would be a good idea but there is also a problem i encountered: when i have many files i worked on, i have trouble reading which files i worked on due to my minor dyslexia... and not showing in the alphabetic or chronological order...
